vcr/api/v1/api.go

```diff
@@ -154,9 +154,61 @@ func (w *Wrapper) Resolve(ctx echo.Context, id string) error {
 		return ctx.NoContent(http.StatusNotFound)
 	}
 
+	// 500
+	if vc == nil && err != nil {
+		return err
+	}
+
+	// transform VC && error
+	result := ResolutionResult{
+		CurrentStatus:        trusted,
+		VerifiableCredential: *vc,
+	}
+
+	switch err {
+	case vcr.ErrUntrusted:
+		result.CurrentStatus = untrusted
+	case vcr.ErrRevoked:
+		result.CurrentStatus = revoked
+	}
+
+	return ctx.JSON(http.StatusOK, result)
+}
+
+func (w *Wrapper) TrustIssuer(ctx echo.Context) error {
+	return changeTrust(ctx, func(cType did.URI, issuer did.URI) error {
+		return w.R.Trust(cType, issuer)
+	})
+}
+
+func (w *Wrapper) UntrustIssuer(ctx echo.Context) error {
+	return changeTrust(ctx, func(cType did.URI, issuer did.URI) error {
+		return w.R.Untrust(cType, issuer)
+	})
+}
+
+type trustChangeFunc func(did.URI, did.URI) error
+
+func changeTrust(ctx echo.Context, f trustChangeFunc) error {
+	var icc = new(CredentialIssuer)
+
+	if err := ctx.Bind(icc); err != nil {
+		return ctx.String(http.StatusBadRequest, fmt.Sprintf("failed to parse request body: %s", err.Error()))
+	}
+
+	d, err := did.ParseURI(icc.Issuer)
 	if err != nil {
+		return ctx.String(http.StatusBadRequest, fmt.Sprintf("failed to parse issuer: %s", err.Error()))
+	}
+
+	cType, err := did.ParseURI(icc.CredentialType)
+	if err != nil {
+		return ctx.String(http.StatusBadRequest, fmt.Sprintf("failed to parse credential type: %s", err.Error()))
+	}
+
+	if err = f(*cType, *d); err != nil {
 		return err
 	}
 
-	return ctx.JSON(http.StatusOK, vc)
+	return ctx.NoContent(http.StatusAccepted)
```

Comment:
huh? An HTTP response code as body of a HTTP response?

Reply:
Ehhh, with `NoContent()` you still have to pass the status code as single arg. Does need to be StatusNoContent though.